raft/raft.go

```diff
@@ -0,0 +1,730 @@
+package raft
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/Dataman-Cloud/swan/log"
+	"github.com/Dataman-Cloud/swan/store"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/coreos/etcd/etcdserver/stats"
+	"github.com/coreos/etcd/pkg/fileutil"
+	"github.com/coreos/etcd/pkg/idutil"
+	"github.com/coreos/etcd/pkg/types"
+	"github.com/coreos/etcd/raft"
+	"github.com/coreos/etcd/raft/raftpb"
+	"github.com/coreos/etcd/rafthttp"
+	"github.com/coreos/etcd/snap"
+	"github.com/coreos/etcd/wal"
+	"github.com/coreos/etcd/wal/walpb"
+	events "github.com/docker/go-events"
+	"github.com/docker/swarmkit/watch"
+	"github.com/pivotal-golang/clock"
+	"golang.org/x/net/context"
+)
+
+const (
+	MaxTransactionBytes = 1.5 * 1024 * 1024
+)
+
+type LeadershipState int
+
+const (
+	// indicates that the node is a raft leader
+	IsLeader LeadershipState = iota
+
+	// indicates that the node is a raft follower
+	IsFollower
+)
+
+var (
+	// returns when an operation was submiited bu the node was stopped in the meantime
+	ErrStopped = errors.New("raft: failed to process the request: node is stopped")
```

Comment:
raft: failed to process the request: node is stopped => raft: failed to process the request: node stopped